src/model_repository_manager.cc

```diff
@@ -419,55 +419,79 @@ ModelRepositoryManager::PollAndUpdate()
 Status
 ModelRepositoryManager::PollAndUpdateInternal(bool* all_models_polled)
```

Comment:
This will allow concurrent poll operations (and also concurrently with explicit model load/unload), and be consistence with `LoadUnloadModels()`. Maybe this is more future proof?

Reply:
The shared utilities (UpdateDependencyGraph(), LoadModelByDependency()) have changed. They used to operate on the model_repository_manager owned graph variable, but now on the provided one.

The unbeneficial change this time is the unlock of `mu_` mutex when the models are loading/unloading.

Reply:
Why not just providing the member variables to those functions then? `UpdateDependencyGraph(..., &dependency_graph_)`

Reply:
Why not just providing the member variables to those functions then? `UpdateDependencyGraph(..., &dependency_graph_)`